### PR TITLE
Feat/guiv2 gallery fields filter sort

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- Add filter menu in grid view
- Add sort menu in grid view
- Field menu for toggling fields
- Cell header icons
- Gallery view  

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [x] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

re #2629

### CODE REVIEW

File: packages/nc-gui-v2/components/smartsheet-header/VirtualCellIcon.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            {{ rollupIcon }}
+          </v-icon>
+          <v-icon v-else-if="relationType === 'bt'" color="info" x-small class="mr-1" v-on="on">
+            {{ rollupIcon }}
+          </v-icon>
+          <v-icon v-else-if="relationType === 'mm'" color="pink" x-small class="mr-1" v-on="on">
+            {{ rollupIcon }}
+          </v-icon>
+        </template>
+* */
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-header/VirtualCellIcon.vue:62]: can this be removed?

File: packages/nc-gui-v2/components/smartsheet-toolbar/FieldsMenu.vue [hunk 2]
```diff
@@ -12,14 +13,27 @@ const { showSystemFields, fieldsOrder, coverImageField, modelValue } = definePro
...
+  () => activeView?.value?.id,
+  async (newVal, oldVal) => {
+    if (newVal !== oldVal && meta?.value) {
+      await loadViewColumns()
+    }
+  },
+  { immediate: true },
+)
+
 /* import draggable from 'vuedraggable'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-toolbar/FieldsMenu.vue:37]: can this be removed?

**Comment by bcakmakoglu** [comment]
@pranavxc We're starting to have a lot of composables.
Might be a good idea to start moving them into subdirectories, by concern, i.e. composables for Spreadsheet etc might go into `composables/spreadsheet` or something. 😄

File: packages/nc-gui-v2/components/smartsheet-header/VirtualCellIcon.vue (hunk 1 above)
Comment (pranavxc) [reply packages/nc-gui-v2/components/smartsheet-header/VirtualCellIcon.vue:62]: yes, just kept for reference

**Review by bcakmakoglu — APPROVED** [review]

**Pull request merged**